base/applications/notepad/settings.c

```diff
@@ -100,9 +101,54 @@ static BOOL QueryBool(HKEY hKey, LPCTSTR pszValueName, BOOL *pbResult)
     return TRUE;
 }
 
-static BOOL QueryString(HKEY hKey, LPCTSTR pszValueName, LPTSTR pszResult, DWORD dwResultSize)
+static BOOL QueryString(HKEY hKey, LPCTSTR pszValueName, LPTSTR pszResult, DWORD dwResultLen)
 {
-    return QueryGeneric(hKey, pszValueName, REG_SZ, pszResult, dwResultSize * sizeof(TCHAR));
+    if (!QueryGeneric(hKey, pszValueName, REG_SZ, pszResult, dwResultLen * sizeof(TCHAR)))
+        return FALSE;
+    assert(dwResultLen > 0);
```

Comment:
Can we bailout here instead (and handle that at the callers sideif needed), instead of adding an assert? That would also avoid the need for the additional include.

Reply:
Deleted `#include <assert.h>`.